pandas/tseries/frequencies.py

```diff
@@ -112,20 +109,16 @@ def infer_freq(
     >>> pd.infer_freq(idx)
     'D'
     """
+    from pandas.api.types import is_datetime64_any_dtype
     from pandas.core.api import DatetimeIndex
 
-    if isinstance(index, ABCSeries):
-        values = index._values
-        if not (
-            lib.is_np_dtype(values.dtype, "mM")
-            or isinstance(values.dtype, DatetimeTZDtype)
-            or values.dtype == object
-        ):
-            raise TypeError(
-                "cannot infer freq from a non-convertible dtype "
-                f"on a Series of {index.dtype}"
-            )
-        index = values
+    if isinstance(index, ABCSeries) and not (
+        is_datetime64_any_dtype(index) or index.dtype == object
```

Comment:
Can you check `dtype.kind == "mM"` on `ArrowDtype` instead?

Reply:
I am a bit surprised by this comment, isn't `pandas.api.types.is_` supposed to be a single source of truth of what is considered a certain data type? How do you remember what are all the conditions to check otherwise?

Reply:
For internal usages, checking `dtype.kind` is more performant than `api.types` functions so that is preferred

Reply:
Ah I guess I accidentally removed checking for timedelta. Although no tests failed. I am adding some tests for timedeltas as well.

Reply:
There is another bug: when trying with `Index["duration[s][pyarrow]"]`, it complains about the property `asi8` missing. asi8 is [denoted as deprecated on general index types](https://github.com/pandas-dev/pandas/blob/8dab54d6573f7186ff0c3b6364d5e4dd635ff3e7/pandas/core/indexes/base.py#L674-L689). Moreover, the `FrequencyInferer` assumes numpy dtype and makes use of internals (like `index._data._ndarray`)

the easiest workaround would be to cast to numpy, but I guess we want to avoid that and keep dtype as pyarrow inside the ops.